### ISSUE TITLE
The system duplicates the ESXi Servers

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#1126]
---

Author Name: **Tomás Abad** (Tomás Abad)
Original Redmine Issue: 2109, http://forge.fusioninventory.org/issues/2109
Original Date: 2013-05-08

---

## The environment

My FusionInventory for GLPI v0.83+2.2 test environment is as following:

* The server
** Ubuntu Server 12.04.2 LTS (Quantal Quetzal) - Architecture AMD64
** GLPI v0.83.8
*** GLPI v0.83.8 Agents
*** Bays Management v1.3.2
*** Data Injection v2.2.4
*** FusionInventory for GLPI v0.83+2.2
**** FusionInventory v0.83+2.2
**** FusionInventory DEPLOY v0.83+2.2
**** FusionInventory INVENTORY v0.83+2.2
**** FusionInventory SNMP v0.83+2.2
*** Print to PDF v0.83.3
*** Reports v1.6.1
*** Shell Commands v1.5.2
*** Web Services v1.3.1

* The agents
** FusionInventory-Agent v2.2.7-1 (for GNU/Linux)

## The bug

For the same ESXi Server the system creates a new computer each time it does an inventory of the server. I have a different computer for each file I attach.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Tomás Abad** (Tomás Abad)
Original Date: 2013-05-10T09:50:42Z

---

I truly believe that there is a bug. Here is more information.

These are my FusionInventory INVENTORY "Import options"

```
Monitors         -> Unique import on serial number
Printers         -> Unique import on serial number
Devices          -> Unique import on serial number
Software         -> Yes
Volumes          -> Yes
Antivirus        -> Yes
Virtual machines -> Yes
Location         -> (Empty)
Group            -> (Empty)
Default status   -> (Empty)

Components
   Processor            -> Yes
   Memory               -> Yes
   Hard Drive           -> Yes
   Network Card         -> Yes
   Virtual network card -> Yes
   Graphics Card        -> Yes
   Soundcard            -> Yes
   Drives               -> Yes
   Network drives       -> Yes
   Controllers          -> Yes

```

These are my "Equipment import and link rules" (ordered from highest to lowest priority)

```
Computer serial + uuid
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: Serial Number is already present in GLPI Yes
      Assert to import: Serial Number exist Yes
      Assert to import: UUID is already present in GLPI Yes
      Assert to import: UUID exist Yes
   Actions:
      FusionInventory link Assigh Link if possible, else import denied

Computer serial
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: Serial Number is already present in GLPI Yes
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assigh Link if possible, else import denied

Computer uuid
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: UUID is already present in GLPI Yes
      Assert to import: UUID exist Yes
   Actions:
      FusionInventory link Assigh Link if possible, else import denied

Computer mac
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: MAC is already present in GLPI Yes
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assigh Link if possible, else import denied

Computer import (by serial + uuid)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: Serial Number exist Yes
      Assert to import: UUID exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Computer import (by serial)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Computer import (by uuid)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: UUID exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Computer import (by mac)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Computer import denied
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Computer
   Actions:
      To be unaware of import Assign Yes

Printer serial
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Printer
      Assert to import: Serial Number is already present in GLPI Yes
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied

Printer mac
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Printer
      Assert to import: MAC is already present in GLPI Yes
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied

Printer import (by serial)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Printer
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Printer import (by mac)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Printer
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

Printer import denied
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is Printer
   Actions:
      To be unaware of import Assign Yes

NetworkEquipment serial
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is NetworkEquipment
      Assert to import: Serial Number is already present in GLPI Yes
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied

NetworkEquipment mac
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is NetworkEquipment
      Assert to import: MAC is already present in GLPI Yes
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied

NetworkEquipment import (by serial)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is NetworkEquipment
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

NetworkEquipment import (by mac)
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is NetworkEquipment
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else create device

NetworkEquipment import denied
   Logical operator:
      and
   Criteria:
      Assert to import: Item type is NetworkEquipment
   Actions:
      To be unaware of import Assign Yes

Find serial in all GLPI
   Logical operator:
      and
   Criteria:
      Assert to import: Serial Number is already present in GLPI Yes
      Assert to import: Serial Number exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied


Find mac in all GLPI
   Logical operator:
      and
   Criteria:
      Assert to import: MAC is already present in GLPI Yes
      Assert to import: MAC exist Yes
   Actions:
      FusionInventory link Assign Link if possible, else import denied

```

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Tomás Abad** (Tomás Abad)
Original Date: 2013-05-10T14:21:23Z

---

But it doesn't happen with VMware ESXi servers. It happens with Windows servers and Ubuntu servers too. Here there are two examples:

 Note: With most of the computers the rules work perfectly.


### With a Windows server.

The server apadako01 doesn't existe into GLPI (FusionInventory-Agent v2.2.7-3 inside).

I make three inventories, +the first one+ (file 'apadako01-windows-server-20130510T1102.xml') is in accordance with rule 'Computer import (by serial + uuid)' and it's correct, +the second one+ (file 'apadako01-windows-server-20130510T1438.xml') is in accordance with rule 'Computer import (by serial + uuid)' and it's wrong because it should be in accordance with rule 'Computer serial + uuid', and +the third one+ (file 'apadako01-windows-server-20130510T1531.xml') is in accordance with rule 'Computer import (by serial + uuid)' and it's wrong because it should be also in accordance with rule 'Computer serial + uuid'.

I have got three different computers inside GLPI instead of only one.

### With a Ubuntu server.

The server web02 doesn't existe into GLPI (FusionInventory-Agent v2.1.10 inside).

I make three inventories, +the first one+ (file 'web02-ubuntu-server-20130510T1457.xml') is in accordance with rule 'Computer import (by serial + uuid)' and it's correct, and +the second one+ (file 'web02-ubuntu-server-20130510T1530.xml') is in accordance with rule 'Computer import (by serial + uuid)' and it's wrong because it should be in accordance with rule 'Computer serial + uuid'.

I have got two different computers inside GLPI instead of only one.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Tomás Abad** (Tomás Abad)
Original Date: 2013-05-13T07:16:08Z

---

These are the 'Rule import logs' for 'apadako01' computers.

```
Computer - ID 3947
   Name: apadako01
   Rule import logs:
      #1
           Date: 2013-05-11 20:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48 
         Module: Network discovery
      #2
           Date: 2013-05-10 11:02
      Rule name: Computer import (by serial + uuid)
          Agent: apadako01.hgcr.sescam.jclm.es-2012-04-23-13-30-41
         Module: Inventory


Computer - ID 3959
   Name: apadako01
   Rule import logs:
      #1
           Date: 2013-05-10 16:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #2
           Date: 2013-05-10 14:38
      Rule name: Computer import (by serial + uuid)
          Agent: apadako01.hgcr.sescam.jclm.es-2012-04-23-13-30-41
         Module: Inventory


Computer - ID 3963 
   Name: apadako01
   Rule import logs:
      #1
           Date: 2013-05-10 15:31
      Rule name: Computer import (by serial + uuid)
          Agent: apadako01.hgcr.sescam.jclm.es-2012-04-23-13-30-41
         Module: Inventory


Computer - ID 3974
   Name: apadako01
   Rule import logs:
      #1
           Date: 2013-05-13 08:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #2
           Date: 2013-05-11 16:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #3
           Date: 2013-05-11 12:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #4
           Date: 2013-05-11 08:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #5
           Date: 2013-05-11 04:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #6
           Date: 2013-05-11 00:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #7
           Date: 2013-05-10 20:01
      Rule name: Find mac in all GLPI
          Agent: hatteria-2013-02-19-12-43-48
         Module: Network discovery
      #8
           Date: 2013-05-10 18:56
      Rule name: Computer import (by serial + uuid)
          Agent: apadako01.hgcr.sescam.jclm.es-2012-04-23-13-30-41
         Module: Inventory


Computer - ID xxxx
   Name: apadako01
   Rule import logs:
      ...

```

The computer 'hatter' has the same behaviour but with the rule 'Computer import (by uuid)' instead of the rule 'Computer import (by serial + uuid)'.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Tomás Abad** (Tomás Abad)
Original Date: 2013-07-18T07:37:21Z

---

It seems this bug is resolved in FusionInventory for GLPI v0.83+2.3.1.
